### PR TITLE
[NO-TICKET] fix weird error.stack bug, add random test coverage

### PR DESCRIPTION
**Pull request by nvms**
## Description of change

was looking through the logs of an unrelated build and saw this:

```bash
Unexpected error in catch block - TypeError: Cannot read properties of undefined (reading 'stack')
```

tracked it down to this. error might not always be an object with a `stack` property, so we shouldn't assume that it is. when `stack` isn't there, this throws inside of the `catch` block.

## How to test

should not see the above error in CI (backend test, under `Start server` job)

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-0

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

**Review by AdamAdHocTeam — COMMENTED** [review]
Two things:

1. I still see the error in the BE test logs (maybe another location).
2. Is it safe the output the whole error to cmd line when stack is null. Or does it makes sense to output the message only?

File: src/lib/apiErrorHandler.js [hunk 1]
```diff
@@ -35,7 +36,8 @@ export const handleError = async (req, res, error, logContext) => {
   if (error instanceof Sequelize.Error) {
     await handleSequelizeError(req, res, error, logContext);
   } else {
-    logger.error(`${logContext.namespace} - UNEXPECTED ERROR - ${error.stack}`);
+    const err = error?.stack || error?.message || error;
```
Comment (thewatermethod) [inline src/lib/apiErrorHandler.js:39]: I don't know if you want to kill the if statement and just log the whole error here as well?

Comment (nvms) [reply src/lib/apiErrorHandler.js]: ah yep, thanks for catching, missed this one (b9b320a)

File: src/lib/cron.js [hunk 2]
```diff
@@ -28,7 +28,7 @@ const runJob = () => {
     return updateGrantsRecipients();
   } catch (error) {
     auditLogger.error(`Error processing HSES file: ${error}`);
-    logger.error(error.stack);
+    logger.error(error);
```
Comment (GarrettEHill) [inline src/lib/cron.js:31]: what not just make auditLogger.error call logger.error?

**Review by GarrettEHill — APPROVED** [review]

**Pull request merged**